x-pack/plugins/observability_solution/inventory/public/pages/inventory_page/index.tsx

```diff
@@ -4,53 +4,25 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import { EuiDataGridSorting } from '@elastic/eui';
-import React from 'react';
-import useEffectOnce from 'react-use/lib/useEffectOnce';
-import { EntityColumnIds, EntityType } from '../../../common/entities';
-import { EntitiesGrid } from '../../components/entities_grid';
-import { useInventorySearchBarContext } from '../../context/inventory_search_bar_context_provider';
-import { useInventoryAbortableAsync } from '../../hooks/use_inventory_abortable_async';
+import React, { useEffect } from 'react';
 import { useInventoryParams } from '../../hooks/use_inventory_params';
+import { useInventorySearchBarContext } from '../../context/inventory_search_bar_context_provider';
 import { useInventoryRouter } from '../../hooks/use_inventory_router';
-import { useKibana } from '../../hooks/use_kibana';
+import { useInventoryState } from '../../hooks/use_inventory_state';
+import { UnifiedInventory } from '../../components/grouped_inventory/unified_inventory';
+import { GroupedInventory } from '../../components/grouped_inventory';
 
 export function InventoryPage() {
-  const { searchBarContentSubject$ } = useInventorySearchBarContext();
-  const {
-    services: { inventoryAPIClient },
-  } = useKibana();
-  const { query } = useInventoryParams('/');
-  const { sortDirection, sortField, pageIndex, kuery, entityTypes } = query;
-
+  const { groupBy } = useInventoryState();
+  const { searchBarContentSubject$, refreshSubject$ } = useInventorySearchBarContext();
   const inventoryRoute = useInventoryRouter();
+  const { query } = useInventoryParams('/');
 
-  const {
-    value = { entities: [] },
-    loading,
-    refresh,
-  } = useInventoryAbortableAsync(
-    ({ signal }) => {
-      return inventoryAPIClient.fetch('GET /internal/inventory/entities', {
-        params: {
-          query: {
-            sortDirection,
-            sortField,
-            entityTypes: entityTypes?.length ? JSON.stringify(entityTypes) : undefined,
-            kuery,
-          },
-        },
-        signal,
-      });
-    },
-    [entityTypes, inventoryAPIClient, kuery, sortDirection, sortField]
-  );
-
-  useEffectOnce(() => {
+  useEffect(() => {
     const searchBarContentSubscription = searchBarContentSubject$.subscribe(
       ({ refresh: isRefresh, ...queryParams }) => {
         if (isRefresh) {
-          refresh();
+          refreshSubject$.next();
```

Comment:
You can ignore the refresh on this component, since it will be handled by each component.

You need to change here https://github.com/Bluefinger/kibana/blob/dffcbd44bfe6816d6339348b58c7a405e958a011/x-pack/plugins/observability_solution/inventory/public/components/search_bar/index.tsx#L61-L71 to fire an refresh update when the button is pushed.